mods/sp/rules/aircraft.yaml

```diff
@@ -613,13 +612,60 @@ MUTQUEEN:
 	Explodes:
 		Weapon: QueenExplode
 		EmptyWeapon: QueenExplode
+	Explodes@AImicroManage:
+		RequiresCondition: AImicroManage
+		Weapon: SpawnEggShrapnelE
+		EmptyWeapon: SpawnEggShrapnelE
 
 MUTBIRD:
 	Inherits@3: MUTQUEEN
 	Tooltip:
 		Name: Seeder Bird
 	-Buildable:
 
+NUKECARRYALL_BOT:
+	Inherits: TRNSPORT
+	Inherits@1: ^AutoTargetAllAssaultMove
+	Buildable:
+		Description: AI Nuke Transport
+		Queue: Air
+		BuildPaletteOrder: 6
+		Prerequisites: ~muair, muhall, ~AImoney
+	Valued:
+		Cost: 1900
+	Health:
+		HP: 18000
+	GivesExperience:
+		Experience: 600
+	Cargo:
+		InitialUnits: hvrtruk3_bot
+	Armament@AIaiming:
+		RequireCondition: !unloading
+		Weapon: AITransportAimingDummyWeapon
+		LocalOffset: 950,0,0
+		ReloadingCondition: unloading
+	RejectsOrders@unloading:
+		Except: Unload
+		RequiresCondition: unloading
+	AttackAircraft:
+		AttackType: Hover
+		FacingTolerance: 128
+	AIDeployHelper:
+		DeployChance : 100
+		DeployTicks : 5
+		UndeployTicks : 30
+	AutoTarget:
+		ScanRadius: 9
+		InitialStanceAI: AttackAnything
+	UpdatesPlayerStatistics:
+		AddToArmyValue: true
+	TransformOnCondition:
```

Comment:
Why do you need this? Why is this actor set up in the way it is?
I get you need AI clones for some actions taken, I use similar tricks along. But transforming is never a good idea and induces more bugs. If you only need additional traits and all of them are conditional, it's better to set up a prerequisite on the player actor triggered by the AI condition and wiring the logics to said prerequisite via `GrantConditionOnPrerequisite`.

If this is being done to disguise the actor as it's standard counterpart if it switches ownership to the player, then you only need to override `Selectable: Class` and nothing more.

Reply:
Ummm, there is already AI condition applied globaly as "AImicroControll" condition for unit, however:

1. transfer ownership for aircraft, if you mean mind controll, is buggy. Airbone will become vechicle when near land instead of landing. This will trigger mind controll weapon and I don't think an airbone unit will get mind controlled because it try to climb up a higher cliff or trying attack ground unit like Devourer (I get related bugs on this unit). So we simply stop all Airbone from getting MC, since PVP never consider this trigger.

2. transform part is also a "have to" for me. I don't want AI use this unit in mass so I add a build restriction for AI, and obviously AI will refuse to rebuild it unless it get killed. So in many way of kill actor-self I guess transform can be more reasonable one even if it is buggy, or new player will consider it strange when unload cargo causing carryall explodes/disappear.

Reply:
3. `Selectable: Class` is a good one, and should be applied in many units I add AI-Only here. but I don't know how to update this branch since it is merged. I probably need to refork it.

Reply:
Still, is there any way to get pre-load unit by not building a new cloned actor?

Reply:
And there is many nasty hack becuase unload is a time taken progress can be can be easy override by other activity, like attack, attack move, assualt attack. Even if doing so the AI air transport is still not very good when unloading position is blocked.

Reply:
>  I don't want AI use this unit in mass so I add a build restriction for AI, and obviously AI will refuse to rebuild it unless it get killed, even if it is unload.

I have a recent alternative solution to that as well, based on https://github.com/AttacqueSuperior/Engine/commit/9defbce5d65553aa5dbeb89dc3b974907a30eb34 - using this trait, you can set up a condition on the player actor to trigger a prerequisite when it owns x amount of (pre)loaded actors of this kind and said prereq can then disable the AI clone. You don't need transforms at all.

SP really needs an engine update not just from mainline ORA/ORA but also from AS/Engine.

Reply:
I mean first, when it is loaded with demotruck or something, AI won't build new. when AI unload, whether the unit itself is dead or alive, it allows to build new. In addtion, I guess AI units number restriction should always be set in `SquadManagerBotModule`. 

I already have an idea about it in fact, conditional prerequisite will help.